keras_nlp/layers/random_deletion.py

```diff
@@ -0,0 +1,137 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import random
+from typing import Any
+from typing import Dict
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class RandomDeletion(keras.layers.Layer):
+    """Augments input by randomly deleting words
+
+    Args:
+        probability: probability of a word being chosen for deletion
+        max_deletions: The maximum number of words to replace
+
+    Examples:
+
+    Basic usage.
+    >>> augmenter = keras_nlp.layers.RandomDeletion(
+    ...     probability = 1,
+    ...     max_deletions = 1,
+    ... )
+    >>> augmenter(["dog dog dog dog dog"])
+    <tf.Tensor: shape=(1,), dtype=string, numpy=array([b'dog dog dog dog'],
+    dtype=object)>
+    """
+
+    def __init__(self, probability, max_deletions, **kwargs) -> None:
+        # Check dtype and provide a default.
+        if "dtype" not in kwargs or kwargs["dtype"] is None:
+            kwargs["dtype"] = tf.int32
+        else:
+            dtype = tf.dtypes.as_dtype(kwargs["dtype"])
+            if not dtype.is_integer and dtype != tf.string:
+                raise ValueError(
+                    "Output dtype must be an integer type of a string. "
+                    f"Received: dtype={dtype}"
+                )
+
+        super().__init__(**kwargs)
+        self.probability = probability
+        self.max_deletions = max_deletions
+
+    def call(self, inputs):
+        """Augments input by randomly deleting words
+
+        Args:
+            inputs: A tensor or nested tensor of strings to augment.
+
+        Returns:
+            A tensor or nested tensor of augmented strings.
+        """
+        # If input is a simple String convert it into a list
+        isString = False
+        if isinstance(inputs, str):
+            inputs = [inputs]
+            isString = True
+        # If input is not a tensor convert it into a tensor
+        if not isinstance(inputs, (tf.Tensor, tf.RaggedTensor)):
+            inputs = tf.convert_to_tensor(inputs)
+            inputs = tf.cast(inputs, tf.string)
+
+        def _map_fn(inputs):
+            scalar_input = inputs.shape.rank == 0
+            if scalar_input:
+                inputs = tf.expand_dims(inputs, 0)
+            ragged_words = tf.strings.split(inputs)
+            # Get the row splits for the ragged tensor
+            row_splits = ragged_words.row_splits.numpy()
+            mask = (
+                tf.random.uniform(ragged_words.flat_values.shape)
+                > self.probability
```

Comment:
the format looks a bit strange, is that formatted by black?

Reply:
@chenmoneygithub Yup this was generated by black